src/macros.rs

```diff
@@ -0,0 +1,183 @@
+
+/// Like [gettext](fn.gettext.html), but allows for formatting.
+#[macro_export]
+macro_rules! gettext {
+    ($($arg:tt)*) => ($crate::gettext(std::fmt::format(format_args!($($arg)*))))
+}
+
+/// Like [dgettext](fn.dgettext.html), but allows for formatting.
+#[macro_export]
+macro_rules! dgettext {
+    ($domain:expr, $string:expr, $($string_names:ident = $string_values:expr),*) => ($crate::dgettext($domain, &format!($string, $( $string_names = $string_values ),*)));
+}
+
+/// Like [dcgettext](fn.gettext.html), but allows for formatting.
+#[macro_export]
+macro_rules! dcgettext {
+    ($domain:expr, $category:expr, $string:expr, $($string_names:ident = $string_values:expr),*,) => ($crate::dcgettext($domain, &format!($string, $( $string_names = $string_values ),*), $category));
+}
```

Comment:
All macros but this one match the API of the functions. Here category is in front of the string instead of behind it, because otherwise parsing would get pretty complex from my understanding.